modules/ixBidAdapter.js

```diff
@@ -466,6 +486,66 @@ function buildRequest(validBidRequests, bidderRequest, impressions, version) {
 
   return requests;
 }
+
+/**
+ * Calculates IX diagnostics values and packages them into an object
+ *
+ * @param {array} validBidRequests  The valid bid requests from prebid
+ * @return {Object} IX diag values for ad units
+ */
+function buildIXDiag(validBidRequests) {
+  var adUnitMap = validBidRequests
```

Comment:
Why is this needed?

Reply:
We want to capture metrics for each ad unit for every unique `transactionId` from the valid bid requests

At a high level, we will be passing these metrics in our bid request for further analytics